build.gradle

```diff
@@ -0,0 +1,89 @@
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'scala'
+
+    apply from: "${rootProject.projectDir}/gradle-scalatest.plugin"
+
+    ext {
+        ////////////////////////////////
+        // Commonly used dependencies //
+        ////////////////////////////////
+
+        mkDepGen = { genOne -> { String... names -> names.collect(genOne) } }
```

Comment:
In cases that we need to switch on scala version, we should be able to do if statements.

https://github.com/apache/kafka/blob/trunk/build.gradle#L237

I'd avoid making new words in general. Ex. the following won't invite other gradle folks to maintain this even if we are saving characters. We should just declare inline, and avoid special build language like `mkDepGen` as zipkin is hopefully not special enough to need its own build vocab. I'd rather optimize for immediate understanding vs saving character count. DRY is something to watch out for, but build files and unit tests have special concerns which IMHO override that in many cases.

On technical notes.. if the goal is to force a specific version for twitter things, I would do that in a way that is more idiomatic to gradle. ex.
```
configurations.all {
    resolutionStrategy {
        eachDependency { DependencyResolveDetails details ->
            if (details.requested.group == 'com.twitter:finagle-foo') {
                details.useVersion "6.26.0"
            }
        }
    }
}
```